tests-ng/plugins/file.py

```diff
@@ -0,0 +1,368 @@
+import grp
+import os
+import pwd
+import stat
+from pathlib import Path
+from typing import Tuple
+
+import pytest
+
+
+class File:
+    """Pytest-facing facade for file metadata operations.
+
+    Provides path-based helpers for checking file metadata such as
+    existence, type, permissions, ownership, etc.
+    """
+
+    def exists(self, path: str | Path) -> bool:
+        """Check if a path exists (any type).
+
+        Args:
+            path: File path.
+
+        Returns:
+            bool: True if the path exists, False otherwise.
+        """
+        return Path(path).exists()
+
+    def is_file(self, path: str | Path) -> bool:
+        """Check if a path is a regular file.
+
+        Does not follow symlinks - symlinks return False even if they point to files.
+
+        Args:
+            path: File path.
+
+        Returns:
+            bool: True if the path is a regular file, False otherwise.
+        """
+        p = Path(path)
+        try:
+            return stat.S_ISREG(p.lstat().st_mode)
+        except (OSError, FileNotFoundError):
+            return False
+
+    def is_dir(self, path: str | Path) -> bool:
+        """Check if a path is a directory.
+
+        Does not follow symlinks - symlinks return False even if they point to directories.
+
+        Args:
+            path: File path.
+
+        Returns:
+            bool: True if the path is a directory, False otherwise.
+        """
+        p = Path(path)
+        try:
+            return stat.S_ISDIR(p.lstat().st_mode)
+        except (OSError, FileNotFoundError):
+            return False
+
+    def is_symlink(
+        self,
+        path: str | Path,
+        target: str | Path | None = None,
+        error_on_broken_symlink: bool = True,
+    ) -> bool:
+        """Check if a path is a symlink, optionally validating the target.
+
+        Args:
+            path: File path.
+            target: Optional target path to validate. If provided, checks that the
+                symlink points to this target. Can be a string or Path object.
+                The comparison handles both relative and absolute paths.
+            error_on_broken_symlink: If True (default), raise an error if the symlink
+                is broken (target doesn't exist). If False, return True for any
+                symlink regardless of whether the target exists.
+
+        Returns:
+            bool: True if the path is a symlink and (if target is provided) points
+                to the specified target, False otherwise.
+
+        Raises:
+            FileNotFoundError: If error_on_broken_symlink is True and the symlink
+                target does not exist.
+        """
+        p = Path(path)
+        if not p.is_symlink():
+            return False
+
+        if target is not None:
```

Comment:
```
if target is None:
  return True
```

Would make the code more readable here :)

Reply:
Will move the explicit "is None" a bit up.

Reply:
Reworked 'is_symlink()' for better readability instead. Even `if target is None` we need to check for broken symlinks.